### DISCUSSION TITLE
Getting error: (2026, 'SSL connection error: The message received was unexpected or badly formatted')

### DISCUSSION THREAD

**Discussion by madsmh** [PyMySQL/mysqlclient#678] [Q&A]
When I ask Django to connect over SSL to a remote MariaDB, I get the above error, by database settings are:

> `DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.mysql',
        'USER': 'xx',
        'HOST': 'xx',
        'PASSWORD': 'xx',
        'NAME': 'xx',
        'PORT': '',
        'OPTIONS': {'ssl': True},
    }
}

furthermore, I have

> MariaDB [(none)]>  SHOW VARIABLES LIKE '%tls%';
+---------------+-------------------------+
| Variable_name | Value                   |
+---------------+-------------------------+
| tls_version   | TLSv1.1,TLSv1.2,TLSv1.3 |
+---------------+-------------------------+

I have not been able to resolve this issue for some time, so any help is appreciated!

**Comment by methane** [comment]
Your client may use too old OpenSSL. But I am not sure because you don't explain almost anything about your environment.

**Reply by madsmh** [reply]
Sorry about that, let me post some details:

Here is my MariaDB status:

> MariaDB [(none)]> status
> --------------
> C:\Program Files\MariaDB 11.2\bin\mysql.exe from 11.2.2-MariaDB, client 15.2 for Win64 (AMD64), source revision 929532a9426d085111c24c63de9c23cc54382259
> 
> Connection id:          98
> Current database:
> Current user:           mads@85.81.43.174
> SSL:                    Cipher in use is ECDHE-RSA-AES256-GCM-SHA384
> Current pager:          stdout
> Using outfile:          ''
> Using delimiter:        ;
> Server:                 MariaDB
> Server version:         10.5.22-MariaDB MariaDB Server
> Protocol version:       10
> Connection:             eccentric.dk via TCP/IP
> Server characterset:    latin1
> Db     characterset:    latin1
> Client characterset:    utf8mb4
> Conn.  characterset:    utf8mb4
> TCP port:               3306
> Uptime:                 2 hours 24 min 5 sec
> 

SSL settings:

> MariaDB [(none)]>  SHOW VARIABLES LIKE '%ssl%';
+---------------------+----------------------------------+
| Variable_name       | Value                            |
+---------------------+----------------------------------+
| have_openssl        | YES                              |
| have_ssl            | YES                              |
| ssl_ca              | /etc/mariadb/chain.pem           |
| ssl_capath          |                                  |
| ssl_cert            | /etc/mariadb/fullchain.pem       |
| ssl_cipher          |                                  |
| ssl_crl             |                                  |
| ssl_crlpath         |                                  |
| ssl_key             | /etc/mariadb/privkey.pem         |
| version_ssl_library | OpenSSL 1.1.1k  FIPS 25 Mar 2021 |
+---------------------+----------------------------------+
10 rows in set (0.036 sec)

and TLS:

> MariaDB [(none)]>  SHOW VARIABLES LIKE '%tls%';
+---------------+-------------------------+
| Variable_name | Value                   |
+---------------+-------------------------+
| tls_version   | TLSv1.1,TLSv1.2,TLSv1.3 |
+---------------+-------------------------+

and here is a list of my installed python packages and versions:

> Package                Version
> ---------------------- -------
> asgiref                3.7.2
> Cython                 3.0.6
> Django                 5.0
> mariadb                1.1.8
> mysql-connector-python 8.2.0
> mysqlclient            2.2.0
> packaging              23.2
> pip                    23.3.1
> protobuf               4.21.12
> sqlparse               0.4.4
> tzdata                 2023.3
> wheel                  0.42.0
> 

Please let me know if there is anything else you need.

Best, Mads
